ecosystem/lava-sdk/src/rpcconsumer/rpcconsumer_server.ts

```diff
@@ -181,63 +191,103 @@ export class RPCConsumerServer {
+            trySetFinalRelayResult(response);
+
+            return response;
+          }
+          const reply = relayResult.reply;
+          if (reply == undefined) {
+            const err = new Error("reply is undefined");
+
+            trySetFinalRelayResult(err);
+
+            return err;
+          }
+
+          // we got here if everything is valid
+          const { expectedBlockHeight, numOfProviders } =
+            this.finalizationConsensus.getExpectedBlockHeight(this.chainParser);
+          const pairingAddressesLen =
+            this.consumerSessionManager.getPairingAddressesLength();
+          const latestBlock = reply.getLatestBlock();
+          this.consumerSessionManager.onSessionDone(
             singleConsumerSession,
-            relayResponse.err
+            latestBlock,
+            chainMessage.getApi().getComputeUnits(),
+            relayResponse.latency,
+            singleConsumerSession.calculateExpectedLatency(relayTimeout),
+            expectedBlockHeight,
+            numOfProviders,
+            pairingAddressesLen,
+            isHangingapi
           );
+
+          trySetFinalRelayResult(relayResult);
+
+          return relayResult;
+        })
+        .catch((err: unknown) => {
           if (err instanceof Error) {
-            Logger.error("failed on session failure %s", err);
+            return err;
           }
-        };
-        if (relayResponse.backoff) {
-          const backOffDuration = BACKOFF_TIME_ON_FAILURE;
-          setTimeout(callSessionFailure, backOffDuration); // call sessionFailure after a delay
-        } else {
-          callSessionFailure();
-        }
-        const relayError: RelayError = {
-          providerAddress: providerPublicAddress,
-          err: relayResponse.err,
-        };
-        return [relayError];
-      }
-      const reply = relayResult.reply;
-      if (reply == undefined) {
-        return new Error("reply is undefined");
-      }
 
-      // we got here if everything is valid
-      const { expectedBlockHeight, numOfProviders } =
-        this.finalizationConsensus.getExpectedBlockHeight(this.chainParser);
-      const pairingAddressesLen =
-        this.consumerSessionManager.getPairingAddressesLength();
-      const latestBlock = reply.getLatestBlock();
-      this.consumerSessionManager.onSessionDone(
-        singleConsumerSession,
-        latestBlock,
-        chainMessage.getApi().getComputeUnits(),
-        relayResponse.latency,
-        singleConsumerSession.calculateExpectedLatency(relayTimeout),
-        expectedBlockHeight,
-        numOfProviders,
-        pairingAddressesLen,
-        isHangingapi
+          return new Error("unsupported error " + err);
+        });
+
+      promises.push(promise);
+    }
+
+    const timeout = new Promise<Error | undefined>((resolve) => {
+      setTimeout(
+        () => resolve(SDKErrors.noResponseTimeout),
+        relayTimeout + secondsToMillis(2)
```

Comment:
there is no need to create another timeout here, relayInner has it's own timeout